### PR TITLE
Delete list hack

### PR DESCRIPTION
**Pull request by tfussell**
We used a List to ensure resources were created in a particular order. This shouldn't be necessary now that we're not using ClusterClass.

Here's the output using `helm template . --values ci/values.yaml`:
```yaml
---
# Source: cluster-openstack/templates/cluster.yaml
apiVersion: cluster.x-k8s.io/v1beta1
kind: Cluster
metadata:
  annotations:
    cluster.giantswarm.io/description: "description"
  labels:
    cluster-apps-operator.giantswarm.io/watching: ""
    app: "cluster-openstack"
    app.kubernetes.io/managed-by: "Helm"
    app.kubernetes.io/version: "0.4.0"
    cluster.x-k8s.io/cluster-name: "testwc"
    giantswarm.io/cluster: "testwc"
    giantswarm.io/organization: "testorg"
    helm.sh/chart: "cluster-openstack-0.4.0"
  name: testwc
  namespace: default
spec:
  controlPlaneRef:
    apiVersion: controlplane.cluster.x-k8s.io/v1beta1
    kind: KubeadmControlPlane
    name: testwc
  infrastructureRef:
    apiVersion: infrastructure.cluster.x-k8s.io/v1alpha4
    kind: OpenStackCluster
    name: testwc
---
# Source: cluster-openstack/templates/kubeadm_config_template.yaml
apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
kind: KubeadmConfigTemplate
metadata:
  labels:
    app: "cluster-openstack"
    app.kubernetes.io/managed-by: "Helm"
    app.kubernetes.io/version: "0.4.0"
    cluster.x-k8s.io/cluster-name: "testwc"
    giantswarm.io/cluster: "testwc"
    giantswarm.io/organization: "testorg"
    helm.sh/chart: "cluster-openstack-0.4.0"
  name: testwc-worker
  namespace: default
spec:
  template:
    spec:
      joinConfiguration:
        nodeRegistration:
          kubeletExtraArgs:
            cloud-provider: external
            node-labels: "giantswarm.io/node-pool=worker"
          name: '{{ local_hostname }}'
      files:
        - path: /etc/ssh/trusted-user-ca-keys.pem
          permissions: "0600"
          # Taken from https://vault.operations.giantswarm.io/v1/ssh/public_key
          content: |
            ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIM4cvZ01fLmO9cJbWUj7sfF+NhECgy+Cl0bazSrZX7sU vault-ca@vault.operations.giantswarm.io
            
        - path: /etc/ssh/sshd_config
          permissions: "0600"
          content: |
            # Use most defaults for sshd configuration.
            Subsystem sftp internal-sftp
            ClientAliveInterval 180
            UseDNS no
            UsePAM yes
            PrintLastLog no # handled by PAM
            PrintMotd no # handled by PAM
            # Non defaults (#100)
            ClientAliveCountMax 2
            PasswordAuthentication no
            TrustedUserCAKeys /etc/ssh/trusted-user-ca-keys.pem
            MaxAuthTries 5
            LoginGraceTime 60
            AllowTcpForwarding no
            AllowAgentForwarding no
            
      postKubeadmCommands:
        - systemctl restart sshd
      users:
        - name: giantswarm
          sudo: ALL=(ALL) NOPASSWD:ALL
---
# Source: cluster-openstack/templates/kubeadm_control_plane.yaml
apiVersion: controlplane.cluster.x-k8s.io/v1beta1
kind: KubeadmControlPlane
metadata:
  labels:
    app: "cluster-openstack"
    app.kubernetes.io/managed-by: "Helm"
    app.kubernetes.io/version: "0.4.0"
    cluster.x-k8s.io/cluster-name: "testwc"
    giantswarm.io/cluster: "testwc"
    giantswarm.io/organization: "testorg"
    helm.sh/chart: "cluster-openstack-0.4.0"
  name: testwc
  namespace: default
spec:
  kubeadmConfigSpec:
    clusterConfiguration:
      apiServer:
        certSANs:
          - "api.testwc.test.gigantic.io"
        extraArgs:
          cloud-provider: external
      controllerManager:
        extraArgs:
          cloud-provider: external
    initConfiguration:
      nodeRegistration:
        kubeletExtraArgs:
          cloud-provider: external
        name: '{{ local_hostname }}'
    joinConfiguration:
      nodeRegistration:
        kubeletExtraArgs:
          cloud-provider: external
        name: '{{ local_hostname }}'
    useExperimentalRetryJoin: true
    files:
          - path: /etc/ssh/trusted-user-ca-keys.pem
            permissions: "0600"
            # Taken from https://vault.operations.giantswarm.io/v1/ssh/public_key
            content: |
              ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIM4cvZ01fLmO9cJbWUj7sfF+NhECgy+Cl0bazSrZX7sU vault-ca@vault.operations.giantswarm.io
              
          - path: /etc/ssh/sshd_config
            permissions: "0600"
            content: |
              # Use most defaults for sshd configuration.
              Subsystem sftp internal-sftp
              ClientAliveInterval 180
              UseDNS no
              UsePAM yes
              PrintLastLog no # handled by PAM
              PrintMotd no # handled by PAM
              # Non defaults (#100)
              ClientAliveCountMax 2
              PasswordAuthentication no
              TrustedUserCAKeys /etc/ssh/trusted-user-ca-keys.pem
              MaxAuthTries 5
              LoginGraceTime 60
              AllowTcpForwarding no
              AllowAgentForwarding no
              
          
          - path: /run/kubeadm/gs-kube-proxy-config.yaml
            permissions: "0600"
            content: |
              ---
              apiVersion: kubeproxy.config.k8s.io/v1alpha1
              kind: KubeProxyConfiguration
              metricsBindAddress: "0.0.0.0:10249"
              
          - path: /run/kubeadm/gs-kube-proxy-patch.sh
            permissions: "0700"
            content: |
              #!/usr/bin/env bash
              
              #
              # (PK) I couldn't find a better/simpler way to conifgure it. See:
              # https://github.com/kubernetes-sigs/cluster-api/issues/4512
              #
              
              set -o errexit
              set -o nounset
              set -o pipefail
              
              readonly dir="/run/kubeadm"
              
              # Run this script only if this is the init node.
              if [[ ! -f ${dir}/kubeadm.yaml ]]; then
                  exit 0
              fi
              
              # Exit fast if already appended.
              if [[ ! -f ${dir}/gs-kube-proxy-config.yaml ]]; then
                  exit 0
              fi
              
              cat ${dir}/gs-kube-proxy-config.yaml >> ${dir}/kubeadm.yaml
              rm ${dir}/gs-kube-proxy-config.yaml
              
    preKubeadmCommands:
          - bash /run/kubeadm/gs-kube-proxy-patch.sh
    postKubeadmCommands:
          - systemctl restart sshd
    users:
          - name: giantswarm
            sudo: ALL=(ALL) NOPASSWD:ALL
  machineTemplate:
    infrastructureRef:
      apiVersion: infrastructure.cluster.x-k8s.io/v1alpha4
      kind: OpenStackMachineTemplate
      name: testwc-control-plane-41444dda
  version: 1.20.9
  replicas: 3
---
# Source: cluster-openstack/templates/machine_deployment.yaml
apiVersion: cluster.x-k8s.io/v1beta1
kind: MachineDeployment
metadata:
  labels:
    app: "cluster-openstack"
    app.kubernetes.io/managed-by: "Helm"
    app.kubernetes.io/version: "0.4.0"
    cluster.x-k8s.io/cluster-name: "testwc"
    giantswarm.io/cluster: "testwc"
    giantswarm.io/organization: "testorg"
    helm.sh/chart: "cluster-openstack-0.4.0"
  name: testwc-worker
  namespace: default
spec:
  clusterName: testwc
  replicas: 5
  selector:
    matchLabels: null
  template:
    metadata:
      labels:
        app: "cluster-openstack"
        app.kubernetes.io/managed-by: "Helm"
        app.kubernetes.io/version: "0.4.0"
        cluster.x-k8s.io/cluster-name: "testwc"
        giantswarm.io/cluster: "testwc"
        giantswarm.io/organization: "testorg"
        helm.sh/chart: "cluster-openstack-0.4.0"
    spec:
      bootstrap:
        configRef:
          apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
          kind: KubeadmConfigTemplate
          name: testwc-worker
      clusterName: testwc
      failureDomain: "az1"
      infrastructureRef:
        apiVersion: infrastructure.cluster.x-k8s.io/v1alpha4
        kind: OpenStackMachineTemplate
        name: testwc-worker-8d572991
      version: 1.20.9
---
# Source: cluster-openstack/templates/machine_health_check.yaml
apiVersion: cluster.x-k8s.io/v1beta1
kind: MachineHealthCheck
metadata:
  labels:
    app: "cluster-openstack"
    app.kubernetes.io/managed-by: "Helm"
    app.kubernetes.io/version: "0.4.0"
    cluster.x-k8s.io/cluster-name: "testwc"
    giantswarm.io/cluster: "testwc"
    giantswarm.io/organization: "testorg"
    helm.sh/chart: "cluster-openstack-0.4.0"
  name: testwc
  namespace: default
spec:
  clusterName: testwc
  maxUnhealthy: 40%
  nodeStartupTimeout: 20m0s
  selector:
    matchLabels:
      cluster.x-k8s.io/cluster-name: testwc
  unhealthyConditions:
  - type: Ready
    status: Unknown
    timeout: 10m0s
  - type: Ready
    status: "False"
    timeout: 10m0s
---
# Source: cluster-openstack/templates/openstack_cluster.yaml
apiVersion: infrastructure.cluster.x-k8s.io/v1alpha4
kind: OpenStackCluster
metadata:
  labels:
    app: "cluster-openstack"
    app.kubernetes.io/managed-by: "Helm"
    app.kubernetes.io/version: "0.4.0"
    cluster.x-k8s.io/cluster-name: "testwc"
    giantswarm.io/cluster: "testwc"
    giantswarm.io/organization: "testorg"
    helm.sh/chart: "cluster-openstack-0.4.0"
  name: testwc
  namespace: default
spec:
  tags:
  - giant_swarm_cluster_testmc_testwc
  cloudName: "openstack"
  controlPlaneAvailabilityZones:
  - "az1"
  - "az2"
  - "az3"
  identityRef:
    name: cloud-config
    kind: Secret
  managedAPIServerLoadBalancer: true
  managedSecurityGroups: true
  nodeCidr: "10.6.0.0/24"
  externalNetworkId: "external-network-id"
  allowAllInClusterTraffic: true
  dnsNameservers:
  - "1.1.1.1"
  - "8.8.8.8"
  bastion:
    enabled: true
    instance:
      flavor: "bastion-flavor"
      image: ""
      rootVolume:
        sourceType: image
        diskSize: 10
        sourceUUID: "bastion-image"
---
# Source: cluster-openstack/templates/openstack_machine_template.yaml
apiVersion: infrastructure.cluster.x-k8s.io/v1alpha4
kind: OpenStackMachineTemplate
metadata:
  # we should not delete existing templates during upgrades
  # since capi needs old templates to be able to rollout machines.
  finalizers:
  - cluster-api-cleaner-openstack.finalizers.giantswarm.io
  labels:
    app: "cluster-openstack"
    app.kubernetes.io/managed-by: "Helm"
    app.kubernetes.io/version: "0.4.0"
    cluster.x-k8s.io/cluster-name: "testwc"
    giantswarm.io/cluster: "testwc"
    giantswarm.io/organization: "testorg"
    helm.sh/chart: "cluster-openstack-0.4.0"
  name: testwc-worker-8d572991
  namespace: default
spec:
  template:
    spec:
      cloudName: "openstack"
      flavor: "worker-flavor"
      identityRef:
        name: cloud-config
        kind: Secret
      image: ""
      rootVolume:
        sourceType: image
        diskSize: 30
        sourceUUID: "worker-image"
---
# Source: cluster-openstack/templates/openstack_machine_template.yaml
apiVersion: infrastructure.cluster.x-k8s.io/v1alpha4
kind: OpenStackMachineTemplate
metadata:
  # we should not delete existing templates during upgrades
  # since capi needs old templates to be able to rollout machines.
  finalizers:
  - cluster-api-cleaner-openstack.finalizers.giantswarm.io
  labels:
    app: "cluster-openstack"
    app.kubernetes.io/managed-by: "Helm"
    app.kubernetes.io/version: "0.4.0"
    cluster.x-k8s.io/cluster-name: "testwc"
    giantswarm.io/cluster: "testwc"
    giantswarm.io/organization: "testorg"
    helm.sh/chart: "cluster-openstack-0.4.0"
  name: testwc-control-plane-41444dda
  namespace: default
spec:
  template:
    spec:
      cloudName: "openstack"
      flavor: "control-plane-flavor"
      identityRef:
        name: cloud-config
        kind: Secret
      image: ""
      rootVolume:
        sourceType: image
        diskSize: 20
        sourceUUID: "control-plane-image"
```

### CODE REVIEW

**Comment by tfussell** [comment]
Tested successfully with tekton.

File: helm/cluster-openstack/templates/machine_deployment.yaml [hunk 1]
```diff
@@ -26,6 +27,6 @@ spec:
       infrastructureRef:
         apiVersion: infrastructure.cluster.x-k8s.io/v1alpha4
         kind: OpenStackMachineTemplate
-        name: {{ include "resource.default.name" $ }}-{{ .class }}-{{ include "osmtRevisionByClass" $ }}
+        name: {{ include "resource.default.name" $ }}-{{ .class }}-{{ include "osmtRevisionByClass" (merge . $) }}
```
Comment (pipo02mix) [inline helm/cluster-openstack/templates/machine_deployment.yaml:30]: why this is needed after this change?

Comment (tfussell) [reply helm/cluster-openstack/templates/machine_deployment.yaml:30]: It's equivalent to this line from the deleted file https://github.com/giantswarm/cluster-openstack/blob/92d148917ec39755d285cb4652ce6c934e433dea/helm/cluster-openstack/templates/list.yaml#L13

**Comment by erkanerol** [comment]
I compared generated files with the old ones. They are identical.

**Review by erkanerol — APPROVED** [review]

**Pull request merged**